packages/ember-glimmer/tests/integration/components/attrs-lookup-test.js

```diff
@@ -57,32 +58,44 @@ moduleFor('Components test: attrs lookup', class extends RenderingTest {
 
   ['@htmlbars should be able to modify a provided attr into local state #11571 / #11559'](assert) {
     let instance;
-
     let FooBarComponent = Component.extend({
       init() {
         this._super(...arguments);
         instance = this;
       },
 
       didReceiveAttrs() {
-        this.set('first', this.attrs.first.toUpperCase());
+        this.set('first', this.getAttr('first').toUpperCase());
       }
     });
     this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template: '{{first}}' });
 
-    this.render(`{{foo-bar first="first attr"}}`);
+    this.render(`{{foo-bar first=firstAttr}}`, {
+      firstAttr: 'first attr'
+    });
 
-    assert.equal(instance.get('first'), 'FIRST ATTR');
+    assert.equal(instance.get('first'), 'FIRST ATTR', 'component lookup uses local state');
     this.assertText('FIRST ATTR');
 
     this.runTask(() => this.rerender());
 
-    assert.equal(instance.get('first'), 'FIRST ATTR');
+    assert.equal(instance.get('first'), 'FIRST ATTR', 'component lookup uses local state during rerender');
+    this.assertText('FIRST ATTR');
+
+    this.runTask(() => set(this.context, 'firstAttr', 'second attr'));
+
+    assert.equal(instance.get('first'), 'SECOND ATTR', 'component lookup uses modified local state');
```

Comment:
This is indeed pretty weird, especially when it works in apps. I don't have time to investigate this right now, but I don't want to block this PR on that. Since this is new stuff we added (i.e. we are not loosing coverage), can you comment out the failing steps for now with a TODO comment pointing to this thread? We can revisit this when we enable this test for Glimmer – re-implementing it in Glimmer will probably teach us why.

LGTM otherwise!